ggshield/verticals/hmsl/secret_manager/hashicorp_vault/api_client.py

```diff
@@ -0,0 +1,204 @@
+import logging
+from typing import Dict, Generator, List, Tuple
+from urllib.parse import urlparse
+
+import requests
+
+from ggshield.verticals.hmsl.secret_manager.hashicorp_vault.exceptions import (
+    VaultForbiddenItemError,
+    VaultInvalidUrlError,
+    VaultNotFoundItemError,
+    VaultPathIsNotADirectoryError,
+)
+from ggshield.verticals.hmsl.secret_manager.hashicorp_vault.models import (
+    VaultKvMount,
+    VaultSecrets,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class VaultAPIClient:
+    """Client to interact with Vault API."""
+
+    def __init__(self, vault_url: str, api_token: str) -> None:
+        self.session = requests.Session()
+        self.session.headers.update({"X-Vault-Token": api_token})
+
+        try:
+            if not vault_url.startswith("http://") and not vault_url.startswith(
+                "https://"
+            ):
+                vault_url = f"http://{vault_url}"
```

Comment:
Shouldn't we default to `https` instead of `http`? Also: the `if` can be moved outside the try...except block

Reply:
True, I changed the default and moved the if